unity/src/main/java/io/delta/unity/UCCatalogManagedClient.java

```diff
@@ -0,0 +1,169 @@
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.internal.annotation.VisibleForTesting;
+import io.delta.kernel.internal.files.ParsedLogData;
+import io.delta.storage.commit.Commit;
+import io.delta.storage.commit.GetCommitsResponse;
+import io.delta.storage.commit.uccommitcoordinator.UCClient;
+import io.delta.storage.commit.uccommitcoordinator.UCCommitCoordinatorException;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Client for interacting with Unity Catalog (UC) catalog-managed Delta tables.
+ *
+ * @see UCClient
+ * @see ResolvedTable
+ */
+@Experimental
+public class UCCatalogManagedClient {
+  private static final Logger logger = LoggerFactory.getLogger(UCCatalogManagedClient.class);
+
+  private final UCClient ucClient;
+
+  public UCCatalogManagedClient(UCClient ucClient) {
+    this.ucClient = Objects.requireNonNull(ucClient, "ucClient is null");
+  }
+
+  /**
+   * Loads a Kernel {@link ResolvedTable} at a specific version.
+   *
+   * @param engine The Delta Kernel {@link Engine} to use for loading the table.
+   * @param ucTableId The Unity Catalog table ID, which is a unique identifier for the table in UC.
+   * @param tablePath The path to the Delta table in the underlying storage system.
+   * @param version The version of the table to load.
+   */
+  public ResolvedTable loadTable(Engine engine, String ucTableId, String tablePath, long version) {
+    Objects.requireNonNull(engine, "engine is null");
+    Objects.requireNonNull(ucTableId, "ucTableId is null");
+    Objects.requireNonNull(tablePath, "tablePath is null");
+    checkArgument(version >= 0, "version must be non-negative");
+
+    logger.info("[{}] Loading table at version {}", ucTableId, version);
+    final GetCommitsResponse response = getRatifiedCommitsFromUC(ucTableId, tablePath, version);
+    validateLoadTableVersionExists(ucTableId, version, response.getLatestTableVersion());
+    final List<ParsedLogData> logData =
+        getSortedKernelLogDataFromRatifiedCommits(ucTableId, response.getCommits());
+
+    return timeOperation(
+        "TableManager.loadTable",
+        ucTableId,
+        () ->
+            TableManager.loadTable(tablePath)
+                .atVersion(version)
+                .withLogData(logData)
+                .build(engine));
+  }
+
+  private GetCommitsResponse getRatifiedCommitsFromUC(
+      String ucTableId, String tablePath, long version) {
+    logger.info(
+        "[{}] Invoking the UCClient to get ratified commits at version {}", ucTableId, version);
+
+    return timeOperation(
+        "UCClient.getCommits",
+        ucTableId,
+        () -> {
+          try {
+            return ucClient.getCommits(
```

Comment:
So this seems like we've gotten away with just one rpc. Do we just rely on this call returning everything we need? Or do we expect the engine will have made the other call earlier (e.g. to get the table path)?

If the latter, maybe we should have a method that wraps that up for them?

Reply:
> Or do we expect the engine will have made the other call earlier (e.g. to get the table path)?

This. The Engine uses the official UC client SDK to resolve the table name -> path, get the credentials, etc, and then must use that information to correctly invoke the `UCCatalogManagedClient.loadTable`

Reply:
SG thanks, whatever is easier for you is fine